### ISSUE TITLE
switch2.tSwitch should be (fixed=true) in Modelica.Electrical.Analog.Examples.SwitchWithArc

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1562]
**Reported by jfrenkel on 1 Sep 2014 20:52 UTC**
In the Example Modelica.Electrical.Analog.Examples.SwitchWithArc is the discrete Variable `switch2.tSwitch` marked as `fixed=false`. However there is no initial Equation for `switch2.tSwitch` because the equation 
```
  when edge(off) then
    tSwitch=time;
  end when;
``` 
is not aktiv in the initial equation system. Hence the equation `switch2.tSwitch=pre(switch2.tSwitch);` is generated. In addition the Variable `switch2.tSwitch` is used in the equation
```
  if on then
    v=Ron*i;
  else
    if quenched then
      i=Goff*v;
    else
      v=min(Vmax, V0 + dVdt*(time - tSwitch))*sign(i);
    end if;
  end if;
```
and because of that part of the initial system. 

I have no clue why Dymola does not report a warning. OpenModelica and SimX report a warning.

OpenModelica:
```
Warning: Assuming fixed start value for the following 1 variables:
         switch2.tSwitch:DISCRETE(start = -9.999999999999999e+59 fixed = false protected = true ) .Modelica.Electrical.Analog.Examples.SwitchWithArc, .Modelica.Electrical.Analog.Ideal.CloserWithArc, .Modelica.SIunits.Time type: Real
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1562

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 8 Sep 2014 11:27 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 6 Jan 2015 17:13 UTC**
Fixed in 2ff539e44ccffc212b3b75caf350aacc624107b9 (maintenance) and in bf5a2a881b13332d25546cbf476d39bcd8082324 (trunk):

In fact "fixed = true" is missing for tSwitch. Fixed this in:

Modelica.Electrical.Analog.Ideal.OpenerWithArc
Modelica.Electrical.Analog.Ideal.CloserWithArc
Modelica.Electrical.Analog.Ideal.ControlledOpenerWithArc
Modelica.Electrical.Analog.Ideal.ControlledCloserWithArc

Note, tSwitch has a start value of -inf. When plotting this value, this is not practical. To my understanding, a start value of tSwitch=0, would also work and I recommend to use it.
